If I don't use `GO111MODULE=off` I get this:
```
go get -u github.com/c4milo/github-release
# special fix for gomobile issues
go get golang.org/x/tools/go/packages
go get golang.org/x/mobile/cmd/gomobile@5d9a3325
go: can only use path@version syntax with 'go get' and 'go install' in module-aware mode
make: *** [Makefile:243: gomobile-install] Error 1
```
But I see that `develop` builds fine: https://ci.status.im/job/status-go/job/manual/833/

--- a/Makefile
+++ b/Makefile
@@ -76,7 +76,9 @@ HELP_FUN = \
 		   }
 
 statusgo: ##@build Build status-go as statusd server
-	go build -mod=vendor -i -o $(GOBIN)/statusd -v -tags '$(BUILD_TAGS)' $(BUILD_FLAGS) ./cmd/statusd
+	go build -mod=vendor -i -v \
+		-tags '$(BUILD_TAGS)' $(BUILD_FLAGS) \
+		-o $(GOBIN)/statusd ./cmd/statusd
 	@echo "Compilation done."
 	@echo "Run \"build/bin/statusd -h\" to view available commands."
 
@@ -94,11 +96,15 @@ statusd-prune-docker-image: ##@statusd-prune Build statusd-prune docker image
 		-t $(STATUSD_PRUNE_IMAGE_NAME):latest
 
 bootnode: ##@build Build discovery v5 bootnode using status-go deps
-	go build -i -o $(GOBIN)/bootnode -v -tags '$(BUILD_TAGS)' $(BUILD_FLAGS) ./cmd/bootnode/
+	go build -i -v \
+		-tags '$(BUILD_TAGS)' $(BUILD_FLAGS) \
+		-o $(GOBIN)/bootnode ./cmd/bootnode/
 	@echo "Compilation done."
 
 node-canary: ##@build Build P2P node canary using status-go deps
-	go build -i -o $(GOBIN)/node-canary -v -tags '$(BUILD_TAGS)' $(BUILD_FLAGS) ./cmd/node-canary/
+	go build -i -v \
+		-tags '$(BUILD_TAGS)' $(BUILD_FLAGS) \
+		-o $(GOBIN)/node-canary ./cmd/node-canary/
 	@echo "Compilation done."
 
 statusgo-cross: statusgo-android statusgo-ios
@@ -108,13 +114,21 @@ statusgo-cross: statusgo-android statusgo-ios
 statusgo-android: ##@cross-compile Build status-go for Android
 	@echo "Building status-go for Android..."
 	gomobile init
-	gomobile bind -v -target=android -ldflags="-s -w" $(BUILD_FLAGS_MOBILE) -o build/bin/statusgo.aar github.com/status-im/status-go/mobile
+	gomobile bind -v \
+		-target=android -ldflags="-s -w" \
+		$(BUILD_FLAGS_MOBILE) \
+		-o build/bin/statusgo.aar \
+		github.com/status-im/status-go/mobile
 	@echo "Android cross compilation done in build/bin/statusgo.aar"
 
 statusgo-ios: ##@cross-compile Build status-go for iOS
 	@echo "Building status-go for iOS..."
 	gomobile init
-	gomobile bind -v -target=ios -ldflags="-s -w" $(BUILD_FLAGS_MOBILE) -o build/bin/Statusgo.framework github.com/status-im/status-go/mobile
+	gomobile bind -v \
+		-target=ios -ldflags="-s -w" \
+		$(BUILD_FLAGS_MOBILE) \
+		-o build/bin/Statusgo.framework \
+		github.com/status-im/status-go/mobile
 	@echo "iOS framework cross compilation done in build/bin/Statusgo.framework"
 
 statusgo-library: ##@cross-compile Build status-go as static library for current platform
@@ -122,7 +136,11 @@ statusgo-library: ##@cross-compile Build status-go as static library for current
 	mkdir -p $(GOBIN)/statusgo-lib
 	go run cmd/library/*.go > $(GOBIN)/statusgo-lib/main.go
 	@echo "Building static library..."
-	go build -buildmode=c-archive -o $(GOBIN)/libstatus.a $(BUILD_FLAGS) $(GOBIN)/statusgo-lib
+	go build \
+		$(BUILD_FLAGS) \
+		-buildmode=c-archive \
+		-o $(GOBIN)/libstatus.a \
+		$(GOBIN)/statusgo-lib
 	@echo "Static library built:"
 	@ls -la $(GOBIN)/libstatus.*
 
@@ -131,7 +149,11 @@ statusgo-shared-library: ##@cross-compile Build status-go as shared library for 
 	mkdir -p $(GOBIN)/statusgo-lib
 	go run cmd/library/*.go > $(GOBIN)/statusgo-lib/main.go
 	@echo "Building shared library..."
-	$(GOBIN_SHARED_LIB_CGO_LDFLAGS) go build -buildmode=c-shared -o $(GOBIN)/libstatus.$(GOBIN_SHARED_LIB_EXT) $(BUILD_FLAGS) $(GOBIN)/statusgo-lib
+	$(GOBIN_SHARED_LIB_CGO_LDFLAGS) go build \
+		$(BUILD_FLAGS) \
+		-buildmode=c-shared \
+		-o $(GOBIN)/libstatus.$(GOBIN_SHARED_LIB_EXT) \
+		$(GOBIN)/statusgo-lib
 	@echo "Shared library built:"
 	@ls -la $(GOBIN)/libstatus.*
 

--- a/Makefile
+++ b/Makefile
@@ -113,7 +113,8 @@ statusgo-cross: statusgo-android statusgo-ios
 
 statusgo-android: ##@cross-compile Build status-go for Android
 	@echo "Building status-go for Android..."
-	gomobile init
+	export GO111MODULE=off; \
+	gomobile init; \
 	gomobile bind -v \
 		-target=android -ldflags="-s -w" \
 		$(BUILD_FLAGS_MOBILE) \
@@ -123,7 +124,8 @@ statusgo-android: ##@cross-compile Build status-go for Android
 
 statusgo-ios: ##@cross-compile Build status-go for iOS
 	@echo "Building status-go for iOS..."
-	gomobile init
+	export GO111MODULE=off; \
+	gomobile init; \
 	gomobile bind -v \
 		-target=ios -ldflags="-s -w" \
 		$(BUILD_FLAGS_MOBILE) \
@@ -202,14 +204,42 @@ endif
 	docker push $(BOOTNODE_IMAGE_NAME):$(DOCKER_IMAGE_CUSTOM_TAG)
 	docker push $(DOCKER_IMAGE_NAME):$(DOCKER_IMAGE_CUSTOM_TAG)
 
-install-os-dependencies:
+setup: ##@setup Install all tools
+setup: setup-build setup-dev tidy
+
+setup-dev: ##@setup Install all necessary tools for development
+setup-dev: install-lint install-mock install-modvendor install-protobuf tidy install-os-deps
+
+setup-build: ##@setup Install all necessary build tools
+setup-build: install-lint install-release install-gomobile
+
+install-os-deps: ##@install Operating System Dependencies
 	_assets/scripts/install_deps.sh
 
-setup-dev: lint-install mock-install modvendor-install gen-install tidy install-os-dependencies ##@other Prepare project for development
+install-gomobile: install-xtools
+install-gomobile: ##@install Go Mobile Build Tools
+	GO111MODULE=on  go install golang.org/x/mobile/cmd/...@5d9a3325
+	GO111MODULE=on  go mod download golang.org/x/exp@ec7cb31e
+	GO111MODULE=off go get -d golang.org/x/mobile/cmd/gobind
 
-setup-build: lint-install release-install gomobile-install ##@other Prepare project for build
+install-lint: ##@install Install Linting Tools
+	GO111MODULE=on go install github.com/golangci/golangci-lint/cmd/golangci-lint@v1.33.0
 
-setup: setup-build setup-dev tidy ##@other Prepare project for development and building
+install-mock: ##@install Install Module Mocking Tools
+	GO111MODULE=on go install github.com/golang/mock/mockgen@v1.4.4
+
+install-modvendor: ##@install Install Module Vendoring Tool
+	GO111MODULE=on go install github.com/goware/modvendor@v0.4.0
+
+install-protobuf: ##@install Install Protobuf Generation Tools
+	GO111MODULE=on go install github.com/kevinburke/go-bindata/go-bindata@v3.13.0
+	GO111MODULE=on go install github.com/golang/protobuf/protoc-gen-go@v1.3.4
+
+install-release: ##@install Install Github Release Tools
+	GO111MODULE=on go install github.com/c4milo/github-release@v1.1.0
+
+install-xtools: ##@install Install Miscellaneous Go Tools
+	GO111MODULE=on go install golang.org/x/tools/go/packages/...@v0.1.5
 
 generate: ##@other Regenerate assets and other auto-generated stuff
 	go generate ./static ./static/mailserver_db_migrations ./t ./multiaccounts/... ./appdatabase/... ./protocol/...
@@ -261,29 +291,6 @@ release: check-existing-release
 	    echo "Aborting." && exit 1; \
 	fi
 
-gomobile-install: xtools-install
-	go get golang.org/x/mobile/cmd/gomobile
-
-release-install:
-	go get -u github.com/c4milo/github-release
-
-gen-install:
-	go get github.com/kevinburke/go-bindata/go-bindata@v3.13.0
-	go get github.com/golang/protobuf/protoc-gen-go@v1.3.4
-
-xtools-install:
-	# special fix for gomobile issues
-	go get golang.org/x/tools/go/packages
-
-modvendor-install:
-	# a tool to vendor non-go files
-	# TODO: switch to original repo when https://github.com/goware/modvendor/pull/13 is merged
-	GO111MODULE=off go get -u github.com/adambabik/modvendor
-
-mock-install: ##@other Install mocking tools
-	# keep in sync with go.mod and github.com/golang/mock
-	go get github.com/golang/mock/mockgen@v1.4.1
-
 mock: ##@other Regenerate mocks
 	mockgen -package=fake         -destination=transactions/fake/mock.go             -source=transactions/fake/txservice.go
 	mockgen -package=status       -destination=services/status/account_mock.go       -source=services/status/service.go
@@ -316,10 +323,6 @@ test-e2e-race: test-e2e ##@tests Run e2e tests with -race flag
 canary-test: node-canary
 	# TODO: uncomment that!
 	#_assets/scripts/canary_test_mailservers.sh ./config/cli/fleet-eth.prod.json
-
-lint-install:
-	@# The following installs a specific version of golangci-lint, which is appropriate for a CI server to avoid different results from build to build
-	curl -sfL https://install.goreleaser.com/github.com/golangci/golangci-lint.sh | BINARY=$(GOLANGCI_BINARY) bash -s -- -d -b $(GOPATH)/bin v1.33.0
 
 lint:
 	@echo "lint"

--- a/_assets/ci/Jenkinsfile.android
+++ b/_assets/ci/Jenkinsfile.android
@@ -31,7 +31,6 @@ pipeline {
   environment {
     /* fix for gomobile complaining about missing packages */
     CGO_ENABLED = "1"
-    GO111MODULE = "off"
     /* Other stuff */
     TARGET      = 'android'
     GOPATH      = "${env.WORKSPACE}"
@@ -54,12 +53,12 @@ pipeline {
     } } } }
 
     stage('Setup') { steps { dir(env.STATUS_PATH) {
-      sh 'make setup-build modvendor-install'
+      sh 'make setup-build install-modvendor'
     } } }
 
     stage('Vendoring check') { steps { dir(env.STATUS_PATH) {
       /* fail build if vendoring hasn't been done */
-      sh 'GO111MODULE=on make vendor'
+      sh 'make vendor'
       sh 'git diff --exit-code --no-color --stat vendor/'
     } } }
 

--- a/_assets/ci/Jenkinsfile.ios
+++ b/_assets/ci/Jenkinsfile.ios
@@ -31,7 +31,6 @@ pipeline {
   environment {
     /* fix for gomobile complaining about missing packages */
     CGO_ENABLED = "1"
-    GO111MODULE = "off"
     /* Other stuff */
     TARGET      = 'ios'
     GOPATH      = "${env.WORKSPACE}"

--- a/_assets/ci/Jenkinsfile.tests
+++ b/_assets/ci/Jenkinsfile.tests
@@ -38,12 +38,13 @@ pipeline {
     } } } }
 
     stage('Setup') { steps { dir(env.STATUS_PATH) {
-      sh 'GO111MODULE=off make setup-build modvendor-install'
+      sh 'make setup-build install-modvendor'
     } } }
 
     stage('Vendoring check') { steps { dir(env.STATUS_PATH) {
-      // fail build if vendoring hasn't been done
-      sh 'GO111MODULE=on make vendor && git diff --exit-code --no-color --stat vendor/'
+      /* fail build if vendoring hasn't been done */
+      sh 'make vendor'
+      sh 'git diff --exit-code --no-color --stat vendor/'
     } } }
 
     stage('Lint') { steps { dir(env.STATUS_PATH) {

--- a/_assets/scripts/install_deps.sh
+++ b/_assets/scripts/install_deps.sh
@@ -1,13 +1,14 @@
 #!/usr/bin/env bash
 
-if [ -x "$(command -v apt)" ]; then
+if [[ -x $(command -v apt) ]]; then
   apt install -y protobuf-compiler jq
-fi
-
-if [ -x "$(command -v pacman)" ]; then
+elif [[ -x $(command -v pacman) ]]; then
   pacman -Sy protobuf jq --noconfirm
-fi
-
-if [ -x "$(command -v brew)" ]; then
+elif [[ -x $(command -v brew) ]]; then
   brew install protobuf jq
+elif [[ -x $(command -v nix-env) ]]; then
+  nix-env -iA nixos.protobuf3_13
+else
+  echo "ERROR: No known package manager found!" >&2
+  exit 1
 fi

--- a/go.mod
+++ b/go.mod
@@ -63,6 +63,7 @@ require (
 	go.uber.org/zap v1.15.0
 	golang.org/x/crypto v0.0.0-20210616213533-5ff15b29337e
 	golang.org/x/image v0.0.0-20210220032944-ac19c3e999fb
+	golang.org/x/tools v0.1.2 // indirect
 	gopkg.in/go-playground/assert.v1 v1.2.1 // indirect
 	gopkg.in/go-playground/validator.v9 v9.31.0
 	gopkg.in/natefinch/lumberjack.v2 v2.0.0

